### PR TITLE
DEVDOCS-4934 + DEVDOCS-5106: [update] Apps Tutorial + Quick Start, env vars etc

### PR DESCRIPTION
**Pull request by slsriehl**
# [DEVDOCS-4934] + [DEVDOCS-5106]

## What changed?
Provide a bulleted list in the present tense
* Update env vars
* Add db:setup script
* Update ngrok references from ngrok_id to ngrok_url
* Update some package versions

## Anything else?
Add related PRs, salient notes, ticket numbers, etc.

ping @bc-chaz

[DEVDOCS-4934]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4934?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEVDOCS-5106]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bc-traciporter — DISMISSED** [review]
Nice work @slsriehl 
Your changes look great.
There may be some other articles that need to be changed as well (ngrok_id to ngrok_url). I remember working on something similar.

**Comment by slsriehl** [comment]
@bc-traciporter Good call. Replaced a couple in subscription foundation and webhooks. [See that commit's diff](https://github.com/bigcommerce/dev-docs/pull/2028/commits/12b375692bc9d0c8b09fda4759ee02bbc518ddee).

**Review by bc-traciporter — APPROVED** [review]
Perfect!  This is precisely what I was thinking. 
Thank you for taking care of this.

File: docs/api-docs/apps/apps-quick-start.mdx [hunk 1]
```diff
@@ -91,19 +94,25 @@ FIRE_API_KEY={firebase key}
...
 MYSQL_HOST={mysql host}
 MYSQL_DATABASE={mysql database name}
 MYSQL_USERNAME={mysql username}
 MYSQL_PASSWORD={mysql password}
 MYSQL_PORT={mysql port *optional*}
+
+# Most users do not need to change this
+ENVIRONMENT=bigcommerce.com
+LOGIN_URL=login.${ENVIRONMENT}
+API_URL=api.${ENVIRONMENT}
```
Comment (bc-chaz) [inline docs/api-docs/apps/apps-quick-start.mdx:107]: these variables are for internal testing purposes only - these should not be listed for public consumption

File: docs/api-docs/apps/apps-quick-start.mdx [hunk 2]
```diff
@@ -49,10 +49,10 @@ Next, register the draft app in the [Developer Portal](https://devtools.bigcomme
 2. Click **Create an app**.
 3. Give the app a name.
 4. Click **Technical**.
-5. Enter the **Auth Callback URL** as `https://{ngrok_id}.ngrok.io/api/auth{:http}`. For example, `https://12345.ngrok.io/api/auth{:http}`. You can get the `ngrok_id` from the terminal that is running `ngrok http 3000{:shell}`.
-6. Enter the **Load Callback URL** as `https://{ngrok_id}.ngrok.io/api/load{:http}`.
-7. Enter the **Uninstall Callback URL** as `https://{ngrok_id}.ngrok.io/api/uninstall{:http}`.
-8. If your app supports multiple users, locate the **App Features** section. Select **Multiple Users**, then enter the **Remove User Callback URL** as `https://{ngrok_id}.ngrok.io/api/remove_user{:http}`.
+5. Enter the **Auth Callback URL** as `https://{ngrok_url}/api/auth{:http}`. For example, `https://12345.ngrok.io/api/auth{:http}`. You can get the `ngrok_url` from the terminal that is running `ngrok http 3000{:shell}`.
```
Comment (bc-chaz) [inline docs/api-docs/apps/apps-quick-start.mdx:52]: note: the sample app readme lists this as `ngrok_address` - we might want to use the same language in both locations, whether it's `url` or `address`

File: docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx [hunk 3]
```diff
@@ -44,16 +44,16 @@ To write a `package.json` file with default values, run `npm init -y` instead. D
 
 ## Install npm packages
 
-1. Install `big-design`, `next`, `react`, `react-dom`, and `styled-components`.
+1. Install `big-design`, `big-design-icons`, `big-design-theme`, `dotenv`, `next`, `react`, `react-dom`, and `styled-components`.
```
Comment (bc-chaz) [inline docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx:47]: `big-design-icons`, `big-design-theme`, and `dotenv` packages are not needed at step 1 - their use doesn't come until later, specifically steps 3 and 4

File: docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx [hunk 4]
```diff
@@ -77,14 +77,33 @@ npm install --save-dev babel-plugin-styled-components @types/node @types/react t
...
-    "start": "next start",
+    "start": "next start -p $PORT",
     "test": "echo \"Error: no test specified\" && exit 1"
   }
 }
 ```
 
 3. Save your changes.
 
+## Add the node and npm engines
```
Comment (bc-chaz) [inline docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx:88]: adding `engines` isn't a requirement of this app - this is more so to aid our development

File: docs/api-docs/apps/tutorials/sample-app-next/step-2-connect-to-bc.mdx [hunk 5]
```diff
@@ -230,7 +230,12 @@ Next.js comes pre-equipped to handle environment variables. It loads environment
 CLIENT_ID={app client id}
 CLIENT_SECRET={app secret}
 
-AUTH_CALLBACK=https://{ngrok_id}.ngrok.io/api/auth
+AUTH_CALLBACK=https://{ngrok_url}/api/auth
+
+# Most users do not need to change this
+ENVIRONMENT=bigcommerce.com
+LOGIN_URL=login.${ENVIRONMENT}
+API_URL=api.${ENVIRONMENT}
```
Comment (bc-chaz) [inline docs/api-docs/apps/tutorials/sample-app-next/step-2-connect-to-bc.mdx:238]: please remove

File: docs/api-docs/apps/apps-quick-start.mdx (hunk 2 above)
Comment (slsriehl) [reply docs/api-docs/apps/apps-quick-start.mdx:52]: I prefer URL.

File: docs/api-docs/apps/apps-quick-start.mdx (hunk 1 above)
Comment (slsriehl) [reply docs/api-docs/apps/apps-quick-start.mdx:107]: `API_URL` is required to use app extensions, so I've retained that value.

File: docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx (hunk 3 above)
Comment (slsriehl) [reply docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx:47]: the development server won't start without dotenv, and the user starts it at the end of step 1. 

additionally, the other big-design package is installed in this step. We don't currently have a step to install packages in step 4, so I'm going to leave this in pending a future rewrite.

File: docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx (hunk 4 above)
Comment (slsriehl) [reply docs/api-docs/apps/tutorials/sample-app-next/step-1-setup.mdx:88]: I'm going to leave this in for now to support the reader who's deploying this to a more complex environment.

**Pull request merged**